scipy/stats/_new_distributions.py

```diff
@@ -0,0 +1,327 @@
+    _x_param = _RealParameter('x', domain=_x_support, typical=('a', 'b'))
+
+    _b_domain.define_parameters(_a_param)
+    _log_b_domain.define_parameters(_log_a_param)
+    _x_support.define_parameters(_a_param, _b_param)
+
+    _parameterizations = [_Parameterization(_log_a_param, _log_b_param),
+                          _Parameterization(_a_param, _b_param)]
+    _variable = _x_param
+
+    def __init__(self, *, a=None, b=None, log_a=None, log_b=None, **kwargs):
+        super().__init__(a=a, b=b, log_a=log_a, log_b=log_b, **kwargs)
+
+    def _process_parameters(self, a=None, b=None, log_a=None, log_b=None, **kwargs):
+        a = np.exp(log_a) if a is None else a
+        b = np.exp(log_b) if b is None else b
+        log_a = np.log(a) if log_a is None else log_a
+        log_b = np.log(b) if log_b is None else log_b
+        kwargs.update(dict(a=a, b=b, log_a=log_a, log_b=log_b))
+        return kwargs
+
+    # def _logpdf_formula(self, x, *, log_a, log_b, **kwargs):
+    #     return -np.log(x) - np.log(log_b - log_a)
+
+    def _pdf_formula(self, x, *, log_a, log_b, **kwargs):
+        return ((log_b - log_a)*x)**-1
+
+    # def _cdf_formula(self, x, *, log_a, log_b, **kwargs):
+    #     return (np.log(x) - log_a)/(log_b - log_a)
+
+    def _moment_raw_formula(self, order, log_a, log_b, **kwargs):
+        if order == 0:
+            return self._one
+        t1 = self._one / (log_b - log_a) / order
+        t2 = np.real(np.exp(_log_diff(order * log_b, order * log_a)))
+        return t1 * t2
+
+
+# currently for testing only
+class _Uniform(ContinuousDistribution):
+    r"""Uniform distribution.
+
+    The probability density function of the uniform distribution is:
+
+    .. math::
+
+        f(x; a, b) = \frac{1}
+                          {b - a}
+
+    """
+
+    _a_domain = _RealDomain(endpoints=(-oo, oo))
+    _b_domain = _RealDomain(endpoints=('a', oo))
+    _x_support = _RealDomain(endpoints=('a', 'b'), inclusive=(False, False))
+
+    _a_param = _RealParameter('a', domain=_a_domain, typical=(1e-3, 0.9))
+    _b_param = _RealParameter('b', domain=_b_domain, typical=(1.1, 1e3))
+    _x_param = _RealParameter('x', domain=_x_support, typical=('a', 'b'))
+
+    _b_domain.define_parameters(_a_param)
+    _x_support.define_parameters(_a_param, _b_param)
+
+    _parameterizations = [_Parameterization(_a_param, _b_param)]
+    _variable = _x_param
+
+    def __init__(self, *, a=None, b=None, **kwargs):
+        super().__init__(a=a, b=b, **kwargs)
+
+    def _process_parameters(self, a=None, b=None, ab=None, **kwargs):
+        ab = b - a
+        kwargs.update(dict(a=a, b=b, ab=ab))
+        return kwargs
+
+    def _pdf_formula(self, x, *, ab, **kwargs):
+        return np.full(x.shape, 1/ab)
+
+    def _icdf_formula(self, x, a, b, ab, **kwargs):
+        return a + ab*x
+
+    def _mode_formula(self, *, a, b, ab, **kwargs):
+        return a + 0.5*ab
+
+
+_docfile = "_new_distribution_docs.json"
```

Comment:
Since I misread the code block below and still have some trouble now figuring out exactly what this code does and why it's like this, could you add a code comment to explain what this code does and why it's written this way? And also how docstring edits are supposed to be made?

Reply:
Will do.

Reply:
Done in next commit.